ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -611,6 +610,27 @@ public void shouldStreamPushQueryWithLimitSync() throws Exception {
     assertThat(streamedQueryResult.isComplete(), is(true));
   }
 
+  @Test
+  public void shouldRetryStreamPushQueryAfterError() throws Exception {
+    // When
+    final StreamedQueryResult streamedQueryResult = client.streamQuery(PUSH_QUERY).get();
```

Comment:
the test i am working on. Some things I need to add are the appropriate configs to use spqs (latest offset, enabled push queryv2 etc). I also need to produce rows to the topic after I start the query like we do in spq functional tests. I had some of this code done, but I messed up something on git so i lost it ;/ but will add it back soon in another commit